crates/noirc_evaluator/src/ssa_refactor/ir/basic_block_views.rs

```diff
@@ -0,0 +1,21 @@
+use super::{
+    basic_block::{BasicBlock, BasicBlockId},
+    instruction::TerminatorInstruction,
+};
+
+/// Get an iterator over all successor ids of the provided basic block.
+pub(crate) fn successors_iter(basic_block: &BasicBlock) -> impl Iterator<Item = BasicBlockId> {
+    let ids = match basic_block.terminator() {
+        TerminatorInstruction::Jmp { destination, .. } => {
+            vec![*destination]
+        }
+        TerminatorInstruction::JmpIf { then_destination, else_destination, .. } => {
+            vec![*then_destination, *else_destination]
```

Comment:
I don't think this is a big improvement over `successors` since we end up having to allocate vectors either way